src/domain/communication/room/room.service.events.ts

```diff
@@ -52,10 +53,19 @@ export class RoomServiceEvents {
     mentions: Mention[],
     question: Pick<IMessage, 'id' | 'message'>,
     agentInfo: AgentInfo,
-    room: IRoom
+    room: IRoom,
+    accessToVirtualContributors: boolean
   ) {
     for (const mention of mentions) {
       if (mention.type === MentionedEntityType.VIRTUAL_CONTRIBUTOR) {
+        // Only throw exception here for the case that there is an actual mention
+        if (!accessToVirtualContributors) {
```

Comment:
Yes we do, as otherwise we are relying on the client logic. If someone sends in a comment mutation with a vc tagged in there the code will respond. The logic has to be enforced server side, not rely on client interaction. The alternative would be to check if the tagged vc is a member of the community (also valid) but this is simpler and actually stronger.